app/javascript/components/supplemental_application/sections/ConfirmedHouseholdIncome.js

```diff
@@ -0,0 +1,122 @@
+import React from 'react'
+import { isNil } from 'lodash'
+
+import FormGrid from '~/components/molecules/FormGrid'
+import FormGroupTextValue from '~/components/atoms/FormGroupTextValue'
+import { RadioGroup, Radio, Text } from 'react-form';
+
+const YesNoRadioGroup = ({field}) => {
+  return (
+    <div className='radio-group-inline'>
+      <RadioGroup field={field}>
+        {(group) => (
+          <React.Fragment>
+            <p className="radio-inline">
+              <Radio group={group} value={`true`} id={`${field}-yes`}/>
+              <label className="radio-inline_label" htmlFor={`${field}-yes`}>Yes</label>
+            </p>
+            <p className="radio-inline">
+              <Radio group={group} value={`false`} id={`${field}-no`}/>
+              <label className="radio-inline_label" htmlFor={`${field}-no`}>No</label>
+            </p>
+          </React.Fragment>
+        )}
+      </RadioGroup>
+    </div>
+  )
+}
+
+const getAMIPercent = (income, ami) => {
+  console.log('Calculating the ami percent', income)
+  if (isNil(income))
+    return "Enter HH Income"
+  if (isNil(ami))
+    return "Error: AMI unknown"
+  return income / ami
+}
+
+
+
+class ConfirmedHouseholdIncome extends React.Component {
+
+  state = {
+    hudAMI: null,
+    stateAMI: null
+  }
+
+  componentDidMount() {
+    // Display a calculated annual income if applicant only provided monthly income
+    if (isNil(this.props.application.annual_income)) {
```

Comment:
@fddayan this is my proposed implementation of updating annual income based on monthly income

Reply:
let me see if I follow te requirements correctly
1. We want `anual_income` to be calculated based from other fields
1. However, we want to let the user to set any free text in `anual_income`. Calculated value will be set only if `anual_income` is not present

I feel we can do all this in the `render` function and there is no need for `componentDidMount` or `componentDidUpdate`. 
```
render():
if annual_income is empty 
  anual_income = dervied_value_from_fancy_fields()
end
```
  
about using set `fromApi.setValue`, yes, I would use that to set the value (...and that will update the input text)